Library/Homebrew/os/mac.rb

```diff
@@ -37,6 +37,15 @@ def outdated_release?
       version < "10.9"
     end
 
+    def os_name(version = @version)
```

Comment:
@MikeMcQuaid use `Version` here because
- we need to pass `Version` to `os_name` on `macos_requirement`
- `"10.7" > "10.12"` so plain string comparison is not sufficient